builder/builder.go

```diff
@@ -62,91 +87,60 @@ func FromBuild(build *api.Build) Builder {
 	}
 }
 
-// NewBuild is the API entry for the BuilderScheduler. Create a new Build instance based on PlatformBuild.
-func NewBuild(platformBuild api.PlatformBuild, publishImage string, buildName string) Scheduler {
-	// TODO: Figure if we need a PlatformBuild builder fluent api.
-	// TODO: Verify structure integrity
+// NewBuild is the API entry for the Builder. Create a new Build instance based on PlatformBuild.
+func NewBuild(info BuilderInfo) Scheduler {
 	ctx := buildContext{
-		BaseImage: platformBuild.Spec.BaseImage,
+		BaseImage: info.Platform.Spec.BaseImage,
 		C:         context.TODO(),
 	}
 
-	if platformBuild.Spec.BuildStrategy == api.BuildStrategyPod && platformBuild.Spec.PublishStrategy == api.PlatformBuildPublishStrategyKaniko {
-		ctx.Build = api.NewKanikoBuild(platformBuild, publishImage, buildName)
-	} else {
-		panic(fmt.Errorf("BuildStrategy %s with PublishStrategy %s is not supported", platformBuild.Spec.BuildStrategy, platformBuild.Spec.PublishStrategy))
+	for _, v := range schedulers {
```

Comment:
In this way, If we have got more than one builder that can handle the build, we will create the scheduler only for the first one. Is this  behavior the expected one?

Reply:
Yes, it's. They should be in priority order, and add an attribute in the `Platform` to clear state what builder to use.

Reply:
isn't better to use a priority queue or a stack?

Reply:
Not at all. This is a simple map with 5 entries tops.